src/utils/filterChange.ts

```diff
@@ -0,0 +1,17 @@
+import { selectRootCores } from "./selectRootCores";
+import { Core, FileType } from "../tree/types";
+
+/**  Will be the MarkDownTree with only Folders
+ * @param {Event} target  The whole MarkDownTree
+ * @returns {Function|null} the Function used for filter
+ */
+
+export const filterChange = (target: any): Function | null => {
+  if (target.checked) {
+    return selectRootCores;
```

Comment:
Don’t you need to () to call it? If not no worries. As long as the test pass.

Reply:
@ralph-dev 
it doesn't execute or call the function. it just checks if the switch is on, and if it so, it will return the function.
then in the main program we call filterChange on onChange of the switch and pass the value as the filter to the generateMarkDownTree
about the test... this needs the React Switch because of the condition "target.checked", so I couldn't really do the test properly.

Reply:
Ok not a problem, your good to merge then